Content/NPCs/Bosses/DripplerBoss/DripplerBoss.cs

```diff
@@ -483,116 +488,99 @@ public override bool StrikeNPC(ref double damage, int defense, ref float knockba
-                }
-                else if (choice == 1) // Warrior
-                {
-                    Item.NewItem((int)npc.position.X, (int)npc.position.Y, npc.width, npc.height, ModContent.ItemType<BloodyTeeth>());
-                }
-                else if (choice == 2) // Ranger
-                {
-                    Item.NewItem((int)npc.position.X, (int)npc.position.Y, npc.width, npc.height, ModContent.ItemType<DripplerEye>());
-                }
-                else if (choice == 3) // Mage
-                {
-                    Item.NewItem((int)npc.position.X, (int)npc.position.Y, npc.width, npc.height, ModContent.ItemType<ShatteredOrb>());
-                }
-
-                int necklaceChance = Main.rand.Next(5);
-                if (necklaceChance == 0)
-                {
-                    Item.NewItem((int)npc.position.X, (int)npc.position.Y, npc.width, npc.height, ItemID.SharkToothNecklace);
-                }
-
-                if (Main.rand.Next(7) == 0)
-                {
-                    Item.NewItem((int)npc.position.X, (int)npc.position.Y, npc.width, npc.height, ModContent.ItemType<DripMask>());
-                }
-
-                Item.NewItem((int)npc.position.X, (int)npc.position.Y, npc.width, npc.height, ModContent.ItemType<CancerInABottle>(), Main.rand.Next(6, 10));
+            Gore.NewGore(NPC.position, NPC.velocity, Mod.Find<ModGore>("Gores/DripplerBoss" + (Main.rand.Next(1, 4)).ToString()).Type, NPC.scale);
+            Gore.NewGore(NPC.position, NPC.velocity, Mod.Find<ModGore>("Gores/DripplerBoss" + (Main.rand.Next(1, 4)).ToString()).Type, NPC.scale);
+        }
 
-                if (Main.rand.Next(10) == 0) // Trophy Dropchance
-                {
-                    Item.NewItem((int)npc.position.X, (int)npc.position.Y, npc.width, npc.height, ModContent.ItemType<DripplerTrophy>());
-                }
-            }
+        public override void ModifyNPCLoot(NPCLoot npcLoot)
+        {
+            npcLoot.Add(ItemDropRule.BossBag(ItemType<DripplerBag>()));
+
+            // If not expert mode, one off SinisterBlood, BloodyTeeth, DripplerEye and ShatteredOrb,
+            // 1/5 chance of shark tooth neclace
+            // 1/7 DripMask
+            // 100% 6-10 CancerInABottle
+            // 1/10 DripplerTrophy
+            var nonExpert = new LeadingConditionRule(new Conditions.NotExpert());
+            nonExpert
+                .OnSuccess(ItemDropRule.OneFromOptions(1,
+                    ItemType<SinisterBlood>(),
+                    ItemType<BloodyTeeth>(),
+                    ItemType<DripplerEye>(),
+                    ItemType<ShatteredOrb>()))
+                .OnSuccess(ItemDropRule.Common(ItemID.SharkToothNecklace, 5))
+                .OnSuccess(ItemDropRule.Common(ItemType<DripMask>(), 7))
+                .OnSuccess(ItemDropRule.Common(ItemType<CancerInABottle>(), 1, 6, 10))
+                .OnSuccess(ItemDropRule.Common(ItemType<DripplerTrophy>(), 10));
```

Comment:
Fancy new system for loot. I expect I've made at least a few errors in here. Should be tested.